code/API_definitions/quality-on-demand.yaml

```
@@ -508,12 +524,15 @@ components:
               example: "c8974e592c2fa383d4a3960714"
               description: Authentication token for callback API
       required:
-        - device
         - applicationServer
         - qosProfile
 
     SessionInfo:
-      description: Session related information.
+      description: |
+        Session related information returned in responses.
+        Within the optional device object only one identifier will be returned (with the original value provided in CreateSession).
```

```suggestion
        Device object only to be returned if provided in createSession, and with the same identifier values that were provided originally.
```
We may provide several identifiers in device in the request, so it would be confusing saying that only one must be returned in that case

---

@jlurien I followed here the proposal from @eric-murray in https://github.com/camaraproject/QualityOnDemand/issues/328#issuecomment-2253079545:

> in fact, I would go further, and say that only one "identifier" (it might be a value pair) should be in the session info, which identifier being decided by the API provider. This is because of https://github.com/camaraproject/Commonalities/issues/259. We cannot now return 422 - DEVICE_IDENTIFIERS_MISMATCH (otherwise the API consumer will learn that a given ipv4address is not being used by a given phoneNumber), so better to accept one and ignore the rest.

So the one returned identifier would be the one used to identify the device, without giving further information about the other identifiers. If we would return unsupported identifiers (e.g. IPv4 address/port value pair) it would be also confusing.

---

Even if we go for Eric's proposal, the current text is ambiguous to me. Is the optional device to be returned if present in createSession request, or is it up to the implementation to return it?

Assuming that implementation considers only one and ignores the rest, I would reinforce that, something like:
```
Optional device object only to be returned if provided in createSession. If more that one type of device identifier was provided, only one identifier will be returned (at implementation choice and with the original value provided in createSession)

```

---

@jlurien I like your proposal and have applied it in [xxx](https://github.com/camaraproject/QualityOnDemand/pull/326/commits/27a767294c8ae8b959bf67f26a795ea09c7fbd30). I deleted also the now redundant sentence from line 464.